h2/config.py

```diff
@@ -25,6 +26,21 @@ def __set__(self, instance, value):
         setattr(instance, self.attr_name, value)
 
 
+class DummyLogger(logging.Logger):
+    """
+    An Logger object that does not actual logging, hence a DummyLogger.
+
+    For the class the log operation is merely a no-op.  The intent is to avoid
+    conditionals being sprinkled throughout the hyper-h2 code for calls to
+    logging functions when no logger is passed into the corresponding object.
+    """
+    def _log(self, *vargs, **kwargs):
+        """
+        No-op logging.
+        """
+        pass
```

Comment:
I'm a bit nervous of this low-level patching, but for the moment I think we'll let it land.